precompile/reward_manager.go

```diff
@@ -0,0 +1,472 @@
+}
+
+// RewardManagerConfig implements the StatefulPrecompileConfig
+// interface while adding in the RewardManager specific precompile config.
+type RewardManagerConfig struct {
+	AllowListConfig
+	UpgradeableConfig
+	InitialRewardConfig *InitialRewardConfig `json:"initialRewardConfig,omitempty"`
+}
+
+func init() {
+	parsed, err := abi.JSON(strings.NewReader(RewardManagerRawABI))
+	if err != nil {
+		panic(err)
+	}
+	RewardManagerABI = parsed
+	RewardManagerPrecompile = createRewardManagerPrecompile(RewardManagerAddress)
+}
+
+// NewRewardManagerConfig returns a config for a network upgrade at [blockTimestamp] that enables
+// RewardManager with the given [admins] and [enableds] as members of the allowlist with [initialConfig] as initial rewards config if specified.
+func NewRewardManagerConfig(blockTimestamp *big.Int, admins []common.Address, enableds []common.Address, initialConfig *InitialRewardConfig) *RewardManagerConfig {
+	return &RewardManagerConfig{
+		AllowListConfig: AllowListConfig{
+			AllowListAdmins:  admins,
+			EnabledAddresses: enableds,
+		},
+		UpgradeableConfig:   UpgradeableConfig{BlockTimestamp: blockTimestamp},
+		InitialRewardConfig: initialConfig,
+	}
+}
+
+// NewDisableRewardManagerConfig returns config for a network upgrade at [blockTimestamp]
+// that disables RewardManager.
+func NewDisableRewardManagerConfig(blockTimestamp *big.Int) *RewardManagerConfig {
+	return &RewardManagerConfig{
+		UpgradeableConfig: UpgradeableConfig{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+
+// Equal returns true if [s] is a [*RewardManagerConfig] and it has been configured identical to [c].
+func (c *RewardManagerConfig) Equal(s StatefulPrecompileConfig) bool {
+	// typecast before comparison
+	other, ok := (s).(*RewardManagerConfig)
+	if !ok {
+		return false
+	}
+	// CUSTOM CODE STARTS HERE
+	// modify this boolean accordingly with your custom RewardManagerConfig, to check if [other] and the current [c] are equal
+	// if RewardManagerConfig contains only UpgradeableConfig and AllowListConfig you can skip modifying it.
+	equals := c.UpgradeableConfig.Equal(&other.UpgradeableConfig) && c.AllowListConfig.Equal(&other.AllowListConfig)
+	if !equals {
+		return false
+	}
+
+	if c.InitialRewardConfig == nil {
+		return other.InitialRewardConfig == nil
+	}
+
+	return c.InitialRewardConfig.Equal(other.InitialRewardConfig)
+}
+
+// Address returns the address of the RewardManager. Addresses reside under the precompile/params.go
+// Select a non-conflicting address and set it in the params.go.
+func (c *RewardManagerConfig) Address() common.Address {
+	return RewardManagerAddress
+}
+
+// Configure configures [state] with the initial configuration.
+func (c *RewardManagerConfig) Configure(chainConfig ChainConfig, state StateDB, _ BlockContext) {
+	c.AllowListConfig.Configure(state, RewardManagerAddress)
+	// CUSTOM CODE STARTS HERE
```

Comment:
Can we remove all `CUSTOM CODE` labels from this? I think we should generally remove these from production code.